helix-vcs/src/git.rs

```diff
@@ -103,3 +340,128 @@ fn find_file_in_commit(repo: &Repository, commit: &Commit, file: &Path) -> Optio
         EntryMode::Blob | EntryMode::BlobExecutable => Some(tree_entry.object_id()),
     }
 }
+
+fn git_meta_from_path(
+    path: &Path,
+    autocrlf: bool,
+) -> Result<Option<(FileEntryMode, ObjectId)>, std::io::Error> {
+    // Windows doesn't support symlinks. This block runs fine but is just wasting CPU cycles.
+    #[cfg(not(windows))]
+    match path.symlink_metadata() {
+        Ok(meta) => {
+            if meta.is_symlink() {
+                let link_content = std::fs::read_link(path)?;
+                let link_content = link_content.to_string_lossy();
+                let link_content = link_content.as_bytes();
+
+                let mut hasher = sha1::Sha1::default();
+                hasher.update(b"blob ");
+                hasher.update(format!("{}", link_content.len()).as_bytes());
+                hasher.update(b"\0");
+                hasher.update(link_content);
+
+                let hash: [u8; 20] = hasher.finalize().into();
+
+                return Ok(Some((FileEntryMode::Link, ObjectId::from(hash))));
+            }
+        }
+        Err(_) => return Ok(None),
+    };
```

Comment:
```suggestion
    // Windows doesn't support symlinks. This block runs fine but is just wasting CPU cycles.
    if cfg!(not(windows)) {
        match path.symlink_metadata() {
            Ok(meta) => {
                if meta.is_symlink() {
                    let link_content = std::fs::read_link(path)?;
                    let link_content = link_content.to_string_lossy();
                    let link_content = link_content.as_bytes();

                    let mut hasher = sha1::Sha1::default();
                    hasher.update(b"blob ");
                    hasher.update(format!("{}", link_content.len()).as_bytes());
                    hasher.update(b"\0");
                    hasher.update(link_content);

                    let hash: [u8; 20] = hasher.finalize().into();

                    return Ok(Some((FileEntryMode::Link, ObjectId::from(hash))));
                }
            }
            Err(_) => return Ok(None),
        }
    }
```

When the code compiles on all platform, I find it better to use `cfg!(...)` because it allows autocompletion everywhere, even when not compiling for the targted feature or arch, unlike `#[cfg(...)]`, so it provides a nicer contributor experience :)

Reply:
The indentation level can be reduced using a tuple match, but I'm not sure it's worth it here